### DISCUSSION TITLE
Report primary theme color via external bus

### DISCUSSION THREAD

**Discussion by balloob** [home-assistant/frontend#11202] [Other feature requests]
Mobile applications need to know about the current primary theme color so that they can color the status bar. We should report this color when loaded, and whenever the theme changes. We should report this via the [external bus](https://developers.home-assistant.io/docs/en/frontend_external_bus.html). The external bus should be able to track this info by listening to events. So we should fire an event when we change the theme.

**Comment by JBassett** [comment]
I don't think primary theme color is always what we want.  What we really need is the header color.  In a lot of themes the primary color doesn't match the header causing us to set the status bar to the "wrong" color.

**Comment by bramkragten** [comment]
This was a change in 0.105, that the header can be separately styled.

**Comment by balloob** [comment]
Should we push the whole theme up via the bus? Or just limit to primary and toolbar color ?

**Comment by bramkragten** [comment]
If the android app would like to style the config panel they might need the entire theme, but should we make that a function that is called when the config panel is opened as it is not used much?
I would only push the primary and toolbar color.

**Comment by robbiet480** [comment]
Just putting in my two cents... if you are going to send the whole theme up, is there any way that you can normalize the parameters to be standard RGB hex colors? I've had issues with this on iOS because sometimes the CSS references other variables and I'm not interested in implementing a whole CSS parser just to make pretty colors. I assume this probably can't happen because frontend also shouldn't need a CSS parser but thought i'd just ask anyway. Thanks
